src/components/content/privacy-wrapper.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  faCubes,
+  faPlayCircle,
+  faSpinner,
+} from '@fortawesome/free-solid-svg-icons'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import React, { ReactChild } from 'react'
+import styled from 'styled-components'
+
+import { instanceData } from '@/data/en'
+import { makeMargin, makePadding, makePrimaryButton } from '@/helper/css'
+
+// inspired by https://github.com/ibrahimcesar/react-lite-youtube-embed
+// also borrowed some code
+
+enum Provider {
+  YouTube = 'youtube',
+  WikimediaCommons = 'wikimedia',
+  Vimeo = 'vimeo',
+  GeoGebra = 'geogebra',
+}
+
+interface PrivacyWrapperProps {
+  children: ReactChild
+  placeholder?: ReactChild
+  type: 'video' | 'applet'
+  provider: Provider
+  previewImageUrl: string
+}
+
+function renderProvider(provider: Provider): string {
+  switch (provider) {
+    case Provider.YouTube:
+      return 'YouTube'
+    case Provider.WikimediaCommons:
+      return 'Wikimedia Commons'
+    case Provider.Vimeo:
+      return 'Vimeo'
+    case Provider.GeoGebra:
+      return 'GeoGebra'
+  }
+}
+
+export function PrivacyWrapper({
+  children,
+  placeholder,
+  type,
+  previewImageUrl,
+  provider,
+}: PrivacyWrapperProps) {
+  const [showIframe, setShowIframe] = React.useState(false)
+
```

Comment:
I removed the stuff with `alreadyConsented`. If we want that, we also need a way for users to withdraw their consent (e.g. by showing a similar message below the video).

Reply:
right, forgot about that. thanks.